### PR TITLE
Fix behavior of get*Status in LocalUnderFileSystem

### PR DESCRIPTION
**Pull request by ZacBlanco**
In the getDirectoryStatus javadoc:
 - If a path is a file, this method should throw an exception

In the getFileStatus javadoc:
 - If a path is a directory, this method should throw an
  exception

Unit tests added for behavior verification.

### CODE REVIEW

**Comment by ZacBlanco** [comment]
@calvinjia this is not major, but I found this minor bug doing some UFS testing.

File: underfs/local/src/test/java/alluxio/underfs/local/LocalUnderFileSystemTest.java [hunk 1]
```diff
@@ -225,6 +227,25 @@ public void renameFile() throws IOException {
...
+    try {
+      UfsDirectoryStatus s = mLocalUfs.getDirectoryStatus(dir);
+      assertTrue(s.isDirectory());
+      assertFalse(s.isFile());
+
+      mLocalUfs.getDirectoryStatus(file);
+      fail("Should have failed to get dir status on file");
+    } catch (IOException e) {
+      // Expect exception
+    }
```
Comment (calvinjia) [inline underfs/local/src/test/java/alluxio/underfs/local/LocalUnderFileSystemTest.java]: Use `ExpectedException`?

Comment (ZacBlanco) [reply underfs/local/src/test/java/alluxio/underfs/local/LocalUnderFileSystemTest.java]: as in the junit rule?

**Comment by ZacBlanco** [comment]
Failure is 
```
Results :

Tests in error: 
alluxio.server.ft.journal.TriggeredCheckpointTest.embeddedJournal(alluxio.server.ft.journal.TriggeredCheckpointTest)
  Run 1: TriggeredCheckpointTest.embeddedJournal:89 » Unknown A recent checkpoint alrea...
  Run 2: TriggeredCheckpointTest.embeddedJournal:89 » Unknown A recent checkpoint alrea...
```

**Comment by ZacBlanco** [comment]
jenkins, test this please

**Comment by calvinjia** [comment]
Jenkins, test this please.

**Comment by calvinjia** [comment]
alluxio-bot, merge this please.

**Comment by calvinjia** [comment]
alluxio-bot, cherry-pick this to branch-2.0 please.

**Pull request merged**